serve/method.go

```diff
@@ -84,17 +147,52 @@ func makeInputJSON(msg *dynamicpb.Message) (string, error) {
 	return string(input), nil
 }
 
-func parseOutputJSON(output string, msg *dynamicpb.Message) error {
-	v := struct {
-		Response json.RawMessage `json:"response"`
-	}{}
+func makeStreamingInputJSON(stream []*dynamicpb.Message) (string, error) {
+	v := request{Stream: make([]json.RawMessage, 0, len(stream))}
+	for _, msg := range stream {
+		b, err := protojson.Marshal(msg)
+		if err != nil {
+			return "", err
+		}
+		v.Stream = append(v.Stream, b)
+	}
+
+	input, err := json.Marshal(&v)
+	if err != nil {
+		return "", err
+	}
+
+	return string(input), nil
+}
+
+func parseOutputJSON(output string, desc protoreflect.MethodDescriptor) ([]*dynamicpb.Message, error) {
+	v := response{}
 	if err := json.Unmarshal([]byte(output), &v); err != nil {
-		return err
+		return nil, err
 	}
 
-	if err := protojson.Unmarshal(v.Response, msg); err != nil {
-		return err
+	var stream []*dynamicpb.Message
+	switch {
+	case desc.IsStreamingServer():
+		stream = make([]*dynamicpb.Message, 0, len(v.Stream))
+		for _, jsonMsg := range v.Stream {
+			msg := dynamicpb.NewMessage(desc.Output())
+			if err := protojson.Unmarshal(jsonMsg, msg); err != nil {
+				return nil, err
+			}
+			stream = append(stream, msg)
+		}
+
+	case v.Response != nil:
+		msg := dynamicpb.NewMessage(desc.Output())
+		if err := protojson.Unmarshal(v.Response, msg); err != nil {
+			return nil, err
+		}
+		stream = append(stream, msg)
+
+	default:
+		return nil, errors.New("Unary method did not return result")
```

Comment:
could still also be client streaming, right? so unary or client streaming did not return result maybe?

Reply:
maybe i have this wrong, getting alil tired.

Reply:
I've redone this method, adding some more validation. I ended up using the phrase "unary server method" for methods that do not stream back from the server. The alternative is "unary or client streaming method" which is a bit long (and I don't like saying "a or b" when the code can know if it is a or b, and I couldn't be bothered making dynamic error messages - they are also harder to grep for).